shell/platform/android/android_shell_holder.h

```diff
@@ -19,17 +20,82 @@
 
 namespace flutter {
 
+//----------------------------------------------------------------------------
+/// @brief      This is the Android owner of the core engine Shell.
+///
+///             This is the top orchestrator class on the C++ side for the
+///             Android embedding. It corresponds to a FlutterEngine on the
+///             Java side. This class is in C++ because the core Shell is in
+///             C++ and an Android specific orchestrator needs to exist to
+///             compose it with other Android specific components such as
+///             the PlatformViewAndroid. This composition of many to one
+///             C++ components would be difficult to do through JNI whereas
+///             a FlutterEngine and AndroidShellHolder has a 1:1 relationship.
+///
+///             Technically, the FlutterJNI class owns this AndroidShellHolder
+///             class instance, but the FlutterJNI class is meant to be mostly
+///             static and has minimal state to perform the C++ pointer <->
+///             Java class instance translation.
+///
 class AndroidShellHolder {
  public:
   AndroidShellHolder(flutter::Settings settings,
                      std::shared_ptr<PlatformViewAndroidJNI> jni_facade,
                      bool is_background_view);
 
+  //----------------------------------------------------------------------------
+  /// @brief      Constructor with its components injected.
+  ///
+  ///             This is similar to the standard constructor, except its
+  ///             members were constructed elsewhere and injected.
+  ///
+  ///             All injected components must be non-null and valid.
+  ///
+  ///             Used when constructing the Shell from the inside out when
+  ///             spawning from an existing Shell.
+  ///
+  AndroidShellHolder(flutter::Settings settings,
```

Comment:
Based on the description this sounds like it could be private?

Reply:
What's the convention for making it visible to make_unique? Didn't find clear patterns. friend std::make_unique?

Reply:
I said private because this talks about it being needed for spawning, if only spawning uses it it can be private.  Otherwise it can be public.

Reply:
I meant I can't make it private and let make_unique invoke this private constructor. Maybe I'm missing something obvious. Haven't done much c++11.

Reply:
Oh wait, I see you problem haha,  I'd just not use make_unique `std::unique_ptr<Foo>(/*private constructor*/new Foo())`

I don't know of any official recommendation for this case.

Reply:
ya, that's what I meant :) don't know what the convention is. Didn't want to bother writing a multi-line regex search. Done